ui/src/app/certificate_requests/row.tsx

```diff
@@ -0,0 +1,32 @@
+const extractCSR = (csrPemString: string) => {
+    //TODO
+}
+
+const extractCert = (certPemString: string) => {
+    //TODO
+}
+
+type rowProps = {
+    id: number,
+    csr: string,
+    certificate: string
+}
+export default function Row({ id, csr, certificate }: rowProps) {
+    return (
+        <tr>
+            <td className="" data-test-column="id">{id}</td>
+            <td className="" data-test-column="details">
+                <p><b>Common Name</b>: example.com</p>
+                <p><b>Subject Alternative Name</b>: example.com, 127.0.0.1, 1.2.3.4.5.56</p>
+            </td>
+            <td className="" data-test-column="status">{certificate == "" ? "outstanding" : (certificate == "rejected" ? "rejected" : "certificate expiry date here")}</td>
+            <td className="" data-test-column="action">
+                <button className="p-button--positive">Sign</button>
+                <button className="p-button--negative">Reject</button>
+            </td>
+            <td className="" data-test-column="delete">
+                <button className="p-button--negative">Delete CSR</button>
```

Comment:
```suggestion
                <button className="p-button--negative">Delete</button>
```

Reply:
This is actually something I was going to raise at some point. Should users be able to reject a CSR after a certificate was uploaded for it? IMO the answer is yes, and through a pebble notice it would remove the cert from the relation.

If so, it feels like using the reject button to delete a certificate may not be clear to users of GoCert. Should we have an extra button to delete a certificate after it was signed? Should we rename `sign` as `approve`? I feel like `sign` implies that users will sign the certificate with gocert, but in reality they're supposed to download the CSR, sign it elsewhere, and upload the matching certificate.

Reply:
You're right that it's two different things, users won't sign the cert with gocert, at least not in the initial implementaiton. I'd suggest to have the two different items:
- **Download CSR**
- **Upload Certificate**

We should also add a way to delete a signed certificate, this item can be named:
- **Revoke Certificate**

Then there is the option to reject the certificate request:
- **Reject Certificate Request**

This would remove the need for Delete, which I think is ambiguous.

Reply:
Clarified the exact items on a team call.